src/libPersistence/IncrementalDB.cpp

```diff
@@ -0,0 +1,465 @@
+/*
+ * Copyright (C) 2019 Zilliqa
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include "IncrementalDB.h"
+#include <list>
+#include "libData/BlockChainData/BlockLinkChain.h"
+#include "libMessage/Messenger.h"
+
+using namespace std;
+
+void IncrementalDB::ChangeDBPointer(const uint64_t& currentDSEpoch,
+                                    const string& dbName) {
+  if (m_DBPointer.at(dbName).first != currentDSEpoch) {
+    m_DBPointer.at(dbName).first = currentDSEpoch;
+    m_DBPointer.at(dbName).second =
+        make_shared<LevelDB>(dbName, m_path, to_string(currentDSEpoch));
+  }
+}
+
+bool IncrementalDB::PutTxBody(const dev::h256& txID, const bytes& body,
+                              const uint64_t& dsEpoch) {
```

Comment:
It should be tx Epoch?

Reply:
No the txns are also sieved according to ds epoch.